components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -62,17 +67,22 @@ class SessionManager(
         }
 
         fun unlink(session: Session) {
-            session.engineSessionHolder.engineObserver?.let { observer ->
-                session.engineSessionHolder.apply {
-                    engineSession?.unregister(observer)
-                    engineSession?.close()
-                    engineSession = null
-                    engineSessionState = null
-                    engineObserver = null
-                }
+            val observer = session.engineSessionHolder.engineObserver
+            val engineSession = session.engineSessionHolder.engineSession
+
+            if (observer != null && engineSession != null) {
+                engineSession.unregister(observer)
             }
 
+            session.engineSessionHolder.engineSession = null
+            session.engineSessionHolder.engineObserver = null
+            session.engineSessionHolder.engineSessionState = null
+
             store?.syncDispatch(UnlinkEngineSessionAction(session.id))
+
+            // Closing the engine session after updating the store so that neither session manager
```

Comment:
nit: I misread this comment a few times: Should this be "Closing the engine session after updating *now* that neither the session manager nor the store keep a reference anymore."?